book/src/pil/types.md

```diff
@@ -0,0 +1,240 @@
+# Types
+
+The powdr-pil language has the following types:
+
+- `bool`
+- `int` (integer)
+- `fe` (field element)
+- `string`
+- tuple
+- array
+- function type
+- expression
+- identity
```

Comment:
I'm wondering, if we will use (`constr` functions](https://github.com/powdr-labs/powdr/issues/960) like:
```
let new_boolean = constr || {
  let x;
  constr x * (1 - x) = 0;
  x
}
```
shouldn't we also call the type `constr`? Also it might generalize over identity in that a `constr` can also be a lookup.